### ISSUE TITLE
Improper use of “alt” attribute for icons

### ISSUE THREAD

**Issue by adriannelee** [cds-snc/notification-planning#587]
Issue group: Misc
Visibility: 2
Evaluation: Partially Supports
Success Criteria: 1.1.1: Non-text Content
Needs design: 

### Description
Icons and images throughout the site should be reviewed for proper use of alt text. An image/icon that is presentational would have alt=“”, others would have appropriate alt text specified. For decorative images, an empty alt text attribute (alt=“”) is recommended over the use role=“presentation” or  aria-hidden=“true”. Do not use role=“presentation” or  aria-hidden=“true” when an alternative text description is present. 

![]()

**Comment by andrewleith** [comment]
Our icons use `<i>` elements which don't have the `alt` attribute.  A review of them all shows we are correctly using `aria-hidden="true"` on all of our icons, assuming that they are all decorative (which I think they are). @amazingphilippe are you able to confirm that?

**Comment by andrewleith** [comment]
For `<img>` tags (which I think this card is actually referencing), I will verify that any with values for their `alt` attributes don't have `aria-hidden="true"`.

**Comment by amazingphilippe** [comment]
I think this issue is about the `<img>` tags that are also icons. I don't have an inventory of them... One example though is the account menu on smaller screens, which shows an icon as an image, but also has some good alt text.

**Comment by amazingphilippe** [comment]
I found a few by searching like this in github: https://github.com/search?q=repo%3Acds-snc%2Fnotification-admin%20%3Cimg&type=code

Basically, we still have a lot of images with `aria-hidden="true"`, but we should probably not do that...

**Comment by andrewleith** [comment]
Yeah just reviewing those now, there are a few with alt tags that have `aria-hidden="true"` so im fixing those.  And some that are missing alt text (like close buttons), so i'm fixing those as well!

For these two, what do you think I should do:
1. https://github.com/cds-snc/notification-admin/blob/main/app/templates/views/email-branding/manage-branding.html#L23
2. https://github.com/cds-snc/notification-admin/blob/main/app/templates/views/service-settings/branding/manage-email-branding.html#L21

Should i just add `alt="logo"` to these? Or just hide it entirely with `alt=""` ?

**Comment by andrewleith** [comment]
I'm not even able to get these pages to show in the app.. i'm not totally sure where they are used, possibly on the platform admin side.  I'm just going to "hide" them for now and we can re-visit if it is an issue during testing.

**Comment by amazingphilippe** [comment]
+1 They might be on the platform admin side. Is there a risk to hide something that we might need to see?

**Comment by andrewleith** [comment]
Small risk I think - they already didn't have alt tags so SR users wouldn't have been getting info about them.  Just looked a little deeper:
- the first one is used in platform admin
- the second is used in a branding request

To play it safe I could use add "logo for service"?  I don't actually have the service name (but could change the code here to pipe it through if you think its necessary)

**Comment by amazingphilippe** [comment]
> used in a branding request

If it is in the form, we could probably use the file name, since they could use this to validate that the right file was uploaded. So to make it more readable, we could do

"Preview of {filename}","Aperçu de {filename}"

**Comment by whabanks** [comment]
QA'd in prod. The only images I saw that did not have proper alt tags were on the [Why GC Notify](https://notification.canada.ca/why-gc-notify) page. It's a GCArticles page so perhaps it's out of scope for this card? Otherwise looks good to me!

**Comment by amazingphilippe** [comment]
I think we can add it to the scope, we can easily edit alt text on GCA. I can assign myself to this

**Comment by amazingphilippe** [comment]
Just went through the images on the Why GC Notify pages: 
The images are decorative, and have the empty `alt=""` attribute set. So looking at the card definition, we should be good to close this!
